service/debugger/debugger.go

```diff
@@ -1922,11 +1953,10 @@ func (d *Debugger) FindLocationSpec(goid int64, frame, deferredCall int, locStr
 
 func (d *Debugger) findLocation(goid int64, frame, deferredCall int, locStr string, locSpec locspec.LocationSpec, includeNonExecutableLines bool, substitutePathRules [][2]string) ([]api.Location, error) {
 	locations := []api.Location{}
-	t := proc.ValidTargets{Group: d.target}
-	for t.Next() {
+	for _, t := range d.target.Targets() {
 		pid := t.Pid()
-		s, _ := proc.ConvertEvalScope(t.Target, goid, frame, deferredCall)
-		locs, err := locSpec.Find(t.Target, d.processArgs, s, locStr, includeNonExecutableLines, substitutePathRules)
+		s, _ := proc.ConvertEvalScope(t, goid, frame, deferredCall)
```

Comment:
This is going to fail if the target isn't valid, it will return a nil eval scope which will get passed down to locSpec.Find, depending on what locSpec is this will make it return an error, which will end up returning an error from the whole function.

This is bad because when we add the exec follow mode some of the children will run for a while and then become invalid processes, when this happens it will be impossible to set any new breakpoints.